### ISSUE TITLE
Merging not working with go-task 2.0

### ISSUE THREAD

**Issue by zbindenren** [go-task/task#112]
**Version 1.4.4**
For:
```bash
$ cat Taskfile.yml
---
task_rhel7: &anchor
  vars:
    VERSION: rhel7
  cmds:
    - echo {{.VERSION}}
task_rhel6:
  <<: *anchor
  vars:
    VERSION: rhel6
```
I got:

```bash
$ task task_rhel7
echo rhel7
rhel7
$ task task_rhel6
echo rhel6
rhel6
```
**Version 2.0.1**

```bash
$ cat Taskfile.yml
---
version: '2'

tasks:
  task_rhel7: &anchor
    vars:
      VERSION: rhel7
    cmds:
      - echo {{.VERSION}}
  task_rhel6:
    <<: *anchor
    vars:
      VERSION: rhel6
```

I get:
```bash
$ task task_rhel7
yaml: unmarshal errors:
  line 13: key "VERSION" already set in map
```

Am I missing something?

**Comment by andreynering** [comment]
Hi @zbindenren 

Odd bug, thanks for reporting. This should be fixed on master.

**Comment by zbindenren** [comment]
Thanks.

**Comment by andreynering** [comment]
@zbindenren I just cherry-picked the fix and released v2.0.2

v2.1 should take some time.

**Comment by zbindenren** [comment]
@andreynering thx
